nucypher/config/node.py

```diff
@@ -301,16 +302,37 @@ def dev_mode(self):
     def known_nodes(self):
         return self.__fleet_state
 
-    def connect_to_blockchain(self, recompile_contracts: bool = False):
+    def connect_to_blockchain(self,
+                              enode: str = None,
+                              recompile_contracts: bool = False,
+                              full_sync: bool = False) -> None:
+        """
+
+        :param enode: ETH seednode or bootnode enode address to start learning from,
+                      i.e. 'enode://e54eebad24dc...e1f6d246bea455@52.71.255.237:30303'
+
+        :param recompile_contracts: Recompile all contracts on connection.
+
+        :return: None
+        """
         if self.federated_only:
             raise NodeConfiguration.ConfigurationError("Cannot connect to blockchain in federated mode")
-
         self.blockchain = Blockchain.connect(provider_uri=self.provider_uri,
                                              compile=recompile_contracts,
-                                             poa=self.poa)
+                                             poa=self.poa,
+                                             fetch_registry=True,
+                                             provider_process=self.provider_process,
+                                             sync=full_sync)
 
+        # Read Ethereum Node Keyring
         self.accounts = self.blockchain.interface.w3.eth.accounts
-        self.log.debug("Established connection to provider {}".format(self.blockchain.interface.provider_uri))
+
+        # Add Ethereum Peer
+        if enode:
+            if self.blockchain.interface.client_version == 'geth':
+                self.blockchain.interface.w3.geth.admin.addPeer(enode)
```

Comment:
this should be changed to a eth.clients.Web3Client method/property

Reply:
100% - There is some outstanding Internal geth process management work to do here, I think best left for a future PR(s).